OpenCover.UI.TestDiscoverer/MSTest/MSTestDiscoverer.cs

```diff
@@ -52,22 +53,46 @@ protected override List<TestClass> DiscoverTestsInAssembly(string dll, AssemblyD
 
                 if (isMSTest)
                 {
-                    var TestClass = new TestClass
+                    addTestClass(dll, type, classes2);
+
+                    if (type.HasNestedTypes) // support for nested [TestClass] element
```

Comment:
Could you turn this in a recursive check? Might not be needed, but you never know that someone writes a test class in a test class in a test class, ... ;-)